src/nashpy/algorithms/lemke_howson_lex.py

```diff
@@ -0,0 +1,174 @@
+"""A class for the Lemke Howson algorithm"""
+import warnings
+from itertools import cycle
+
+import numpy as np
+
+from nashpy.integer_pivoting import (
+    make_tableau,
+    non_basic_variables,
+    zero_basic_variables,
+    pivot_tableau_lex,
+)
+
+
+def shift_tableau(tableau, shape):
+    """
+	Shift a tableau to ensure labels of pairs of tableaux coincide
+
+	Parameters
+	----------
+
+		tableau: a numpy array
+		shape: a tuple
+
+	Returns
+	-------
+
+		tableau: a numpy array
+	"""
+    return np.append(
+        np.roll(tableau[:, :-1], shape[0], axis=1),
+        np.ones((shape[0], 1)),
+        axis=1,
+    )
+
+
+def tableau_to_strategy_lex(tableau, basic_labels, strategy_labels):
+    """
+	Return a strategy vector from a tableau
+
+	Parameters
+	----------
+
+		tableau: a numpy array
+		basic_labels: a set
+		strategy_labels: a set
+
+	Returns
+	-------
+
+		strategy: a numpy array
+	"""
+    vertex = []
+    for column in strategy_labels:
+        if column in basic_labels:
+            for i, row in enumerate(tableau[:, column]):
+                if row != 0:
+                    vertex.append(tableau[i, -1] / row)
+        else:
+            vertex.append(0)
+    strategy = np.array(vertex)
+    return np.divide(strategy, sum(strategy))
+
+
+def lemke_howson_lex(A, B, initial_dropped_label=0):
+    """
+	Obtain the Nash equilibria using the Lemke Howson algorithm implemented
+	using integer pivoting.
+
+	1. Start at the artificial equilibrium (which is fully labeled)
+	2. Choose an initial label to drop and move in the polytope for which
+	   the vertex has that label to the edge
+	   that does not share that label. (This is implemented using integer
+	   pivoting)
+	3. A label will now be duplicated in the other polytope, drop it in a
+	   similar way.
+	4. Repeat steps 2 and 3 until have Nash Equilibrium.
+
+	Parameters
+	----------
+
+		initial_dropped_label: int
+
+	Returns
+	-------
+
+		equilibria: A tuple.
+
+	========
+
+		* Edited such that zero basic variables are considered in nash equilibrium 
+		and integer pivoting is done using lexicographical ordering
+
+	"""
+
+    if np.min(A) <= 0:
```

Comment:
There is a lot of duplication in this function, could you modularise it please. You might need to modify `lemke_howson.lemke_howson` to take a pivoting function as an argument.

Reply:
Can I know how you would like lemke_howson to be modularised? Because other than the pivoting function I also added slack variables, changed how the tableaux cycle and how the strategies are calculated. I'm just afraid it might get too messy if I have to change lemke_howson to accommodate these changes

Reply:
And is it ok of my version terminates with an error if the inputs can’t be handled properly? Because yours gives a runtime warning and still outputs things, so i was wondering if that would be better

Reply:
Because of the above issue with the errors I've put off adding the test for a degenerate game from test_lemke_howson to test_lemke_howson_lex as the runtime warning doesn't happen. 

Thanks for all the help so far!

Reply:
> Can I know how you would like lemke_howson to be modularised? Because other than the pivoting function I also added slack variables, changed how the tableaux cycle and how the strategies are calculated. I'm just afraid it might get too messy if I have to change lemke_howson to accommodate these changes

I plan on looking through this in more detail on Wednesday. It might be that this gets refactored subsequently.

> And is it ok of my version terminates with an error if the inputs can’t be handled properly? Because yours gives a runtime warning and still outputs things, so i was wondering if that would be better

I'd rather the runtime warning, it's important to stay consistent otherwise this leads to confusion. Changing the overall behaviour could be a conversation for another time.

This will also allow you to add those tests.

As I mentioned above, I've got more time on Wednesday so I'll get to this properly then, in the meantime if you could make the changes I've suggested regarding the runtime warning that would be great. Note also that there is a merge conflict with the main branch, if you could fix that that would be great (amongst other things it'll add a coverage report to this PR). Let me know if I can assist with anything.

Reply:
> Note also that there is a merge conflict with the main branch, if you could fix that that would be great (amongst other things it'll add a coverage report to this PR). Let me know if I can assist with anything.

To be clear: I'm asking you to rebase/merge in the current master branch. Let me know if I can assist.

Reply:
Hi sorry but I’ll only be able to access my computer earliest on Friday night, so I won’t be able to make the changes until then.

Reply:
No problem, on rush :) :+1: